test/jdk/java/nio/channels/Channels/TransferTo.java

```diff
@@ -143,9 +146,21 @@ public void testMoreThanTwoGB() throws IOException {
             // preparing two temporary files which will be compared at the end of the test
             Path targetFile = Files.createTempFile("test2GBtarget", null);
             try {
-                // writing 3 GB of random bytes into source file
-                for (int i = 0; i < NUM_WRITES; i++)
-                    Files.write(sourceFile, createRandomBytes(BYTES_PER_WRITE, 0), StandardOpenOption.APPEND);
+                // create (hopefully sparse) file less than 2GB in size
+                final long initLen = (2*1024 - 1)*BYTES_PER_WRITE;
```

Comment:
Perhaps consider clarifying the comment for the initLen calculation 

I do suspect the use of RandomAccessFile improves the throughput here

Reply:
I shall clean up the constants in the next revision.

A crude measurement showed that the execution time of `testMoreThanTwoGB()` was cut in half by this change. This sub-test is the slowest `@Test` in the test.

Reply:
The general approach looks okay but the RAF to open/setLength/close is not needed. Instead change the FC.open to use WRITE and use fc.position(initLen) to set the initial position before writing.

Reply:
Of course; will do.